bionic/flow.py

```diff
@@ -1570,9 +1585,10 @@ def __init__(self, state, _official=False):
                 "use one of the classmethod constructors"
             )
 
-        self._uuid = str(uuid4())
-        self._state = state
-        self._deriver = EntityDeriver(state, self._uuid)
+        uuid = str(uuid4())
+        deriver = EntityDeriver(state, uuid)
```

Comment:
WDYT about creating the `EntityDeriver` and `Cache` inside `_initialize` so we don't have to pass them?

Reply:
It makes the method much simpler. `EntityDeriver` and `Cache` are re-created, but there is little cost in doing so.